src/main/scala/com/target/data_validator/validator/ColumnBased.scala

```diff
@@ -36,8 +57,11 @@ case class MinNumRows(minNumRows: Long) extends ColumnBased("", ValidatorBase.L0
 
   override def quickCheck(row: Row, count: Long, idx: Int): Boolean = {
     failed = count < minNumRows
+    val pctError = if (failed) calculatePctError(minNumRows, count) else "0.00%"
     addEvent(ValidatorCounter("rowCount", count))
-    addEvent(ValidatorCheckEvent(failed, s"MinNumRowCheck $minNumRows ", count, 1))
+    val msg = s"MinNumRowsCheck Expected: ${minNumRows} Actual: ${count} Error %: ${pctError}"
+    val data = LinkedHashMap("expected" -> minNumRows.toString, "actual" -> count.toString, "error_percent" -> pctError)
+    addEvent(ColumnBasedValidatorCheckEvent(failed, data.toMap, msg))
```

Comment:
@colindean I am using `LinkedHashMap` to ensure that the retrieval order is the same as the insertion order. Not that it is absolutely necessary, but I wanted these fields to be written to the json report file in the same order (expected, actual, error_percent), instead of some random different order for each check.

Reply:
That's a good enough reason for me!